lib/oneboxer.rb

```diff
@@ -99,30 +99,55 @@ def self.apply(string_or_doc, extra_paths: nil)
 
     each_onebox_link(doc, extra_paths: extra_paths) do |url, element|
       onebox, _ = yield(url, element)
+      next if onebox.blank?
+
+      parsed_onebox = Nokogiri::HTML5::fragment(onebox)
+      next if parsed_onebox.children.blank?
+
+      changed = true
```

Comment:
Is this used anywhere?

Reply:
Yes, it is used in `PostAnalyzer#cook` to update `cooked` attribute of a post only if at least one Onebox was applied. It is basically an optimization to avoid a call to `.to_html`. I am not sure if the extra complexity is worth though. :thinking: